### ISSUE TITLE
Create headless libwasmer (without the wasmer-artifact, only the static-artifact)

### ISSUE THREAD

**Issue by syrusakbary** [wasmerio/wasmer#3086]
Right now a command already exists (for creating the binary).

We may need to copy/"adapt it" to create : `make build-wasmer-headless-minimal` (binary) -> `make build-capi-headless-static` .

Goal: get into a very minimal `libwasmer.a` (or `wasmer.dll`) that only includes the necessary things to be able to deserialize from a static object

**Comment by syrusakbary** [comment]
We don't need `Engine::deserialize`, we just need `Engine::deserialize_object`

**Comment by epilys** [comment]
Closed/implemented in #3101 

Analyzed compiled `./tests/wasi-wast/wasi/snapshot1/hello.wasm` with <https://github.com/google/bloaty>
<details>
<summary>
`hello` binary with full `libwasmer.a`: (52.2% or 8.65MiB is text)
</summary>

```shell
% bloaty hello-full 
    FILE SIZE        VM SIZE    
 --------------  -------------- 
  52.2%  8.65Mi  52.1%  8.65Mi    __TEXT,__text
  27.7%  4.59Mi  27.6%  4.59Mi    String Table
   6.3%  1.04Mi   6.3%  1.04Mi    Symbol Table
   4.4%   748Ki   4.5%   759Ki    [__LINKEDIT]
   3.2%   539Ki   3.2%   539Ki    __TEXT,__const
   2.7%   450Ki   2.7%   450Ki    __DATA_CONST,__const
   1.4%   245Ki   1.4%   245Ki    __TEXT,__gcc_except_tab
   0.8%   131Ki   0.8%   131Ki    Code Signature
   0.7%   119Ki   0.7%   119Ki    __TEXT,__unwind_info
   0.3%  49.0Ki   0.3%  49.0Ki    Function Start Addresses
   0.2%  39.3Ki   0.2%  39.3Ki    __DATA,__data
   0.1%  11.5Ki   0.1%  11.5Ki    [__DATA_CONST]
   0.0%  8.09Ki   0.0%  6.24Ki    [__DATA]
   0.0%  7.30Ki   0.0%  7.41Ki    [__TEXT]
   0.0%  2.23Ki   0.0%  2.23Ki    __TEXT,__stubs
   0.0%  1.91Ki   0.0%  1.81Ki    [Mach-O Headers]
   0.0%  1.51Ki   0.0%  1.51Ki    __DATA_CONST,__got
   0.0%  1.50Ki   0.0%  1.50Ki    Indirect Symbol Table
   0.0%  1.20Ki   0.0%  1.41Ki    [5 Others]
   0.0%       0   0.0%  1.06Ki    __DATA,__thread_bss
   0.0%       0   0.0%     584    __DATA,__bss
 100.0%  16.6Mi 100.0%  16.6Mi    TOTAL
```
</details>

<details>
<summary>
`hello` binary with headless `libwasmer.a`: (47.3% is 3.73MiB is text)
</summary>

```shell
manos ~/wasmer % bloaty hello-headless 
    FILE SIZE        VM SIZE    
 --------------  -------------- 
  47.3%  3.73Mi  47.3%  3.73Mi    __TEXT,__text
  30.7%  2.42Mi  30.7%  2.42Mi    String Table
   7.3%   587Ki   7.3%   587Ki    Symbol Table
   5.4%   439Ki   5.5%   447Ki    [__LINKEDIT]
   3.7%   295Ki   3.7%   295Ki    __TEXT,__const
   3.3%   265Ki   3.3%   265Ki    __DATA_CONST,__const
   0.8%  62.7Ki   0.8%  62.7Ki    Code Signature
   0.5%  39.3Ki   0.5%  39.3Ki    __DATA,__data
   0.3%  24.4Ki   0.3%  24.4Ki    Function Start Addresses
   0.3%  22.3Ki   0.3%  22.3Ki    __TEXT,__unwind_info
   0.1%  12.0Ki   0.2%  12.1Ki    [__TEXT]
   0.1%  8.30Ki   0.1%  7.37Ki    [__DATA]
   0.1%  7.20Ki   0.1%  7.20Ki    __TEXT,__gcc_except_tab
   0.1%  4.80Ki   0.1%  4.80Ki    [__DATA_CONST]
   0.0%  2.11Ki   0.0%  2.11Ki    __TEXT,__stubs
   0.0%  1.91Ki   0.0%  1.81Ki    [Mach-O Headers]
   0.0%  1.43Ki   0.0%  1.43Ki    __DATA_CONST,__got
   0.0%  1.42Ki   0.0%  1.42Ki    Indirect Symbol Table
   0.0%     492   0.0%     838    [5 Others]
   0.0%       0   0.0%     608    __DATA,__bss
   0.0%     504   0.0%     504    __TEXT,__cstring
 100.0%  7.88Mi 100.0%  7.89Mi    TOTAL
```
</details>

Here are the differences after using `strip`:

```shell
% du -sh hello-*    
 17M    hello-full
 11M    hello-full-strip
7.9M    hello-headless
4.9M    hello-headless-strip
```

`bloaty hello-headless-strip` still reports `3.73MiB` of text:

```shell
 % bloaty hello-headless-strip 
    FILE SIZE        VM SIZE    
 --------------  -------------- 
  76.5%  3.73Mi  76.5%  3.73Mi    __TEXT,__text
```

Inspecting with `xxd hello-headless-strip` and manually counting the segments with text (format strings, messages etc) got me a similar figure (~5MiB because I was overcounting).

## Why is wasmer-headless much smaller then?

It's not even working:

```shell
manos ~ % ~/Desktop/wasmer/bin/wasmer compile qjs.wasm -o compiled.wasmu
Compiler: cranelift
Target: aarch64-apple-darwin
✔ File compiled successfully to `compiled.wasmu`.
manos ~ % ~/Desktop/wasmer/bin/wasmer-headless run compiled.wasmu       
zsh: floating point exception  ~/Desktop/wasmer/bin/wasmer-headless run compiled.wasmu
manos ~ % ~/Desktop/wasmer/bin/wasmer-headless run compiled.wasmu
zsh: segmentation fault  ~/Desktop/wasmer/bin/wasmer-headless run compiled.wasmu
```
